weldx/measurement.py

```diff
@@ -37,71 +28,684 @@ class Signal:
+        data :
+            A set of measurement data that is associated with the output signal of the
+            transformation
+        input_signal_source :
+            The source of the signal that should be used as input of the transformation.
+            If `None` is provided, the name of the last added transformation (or the
+            source, if no transformation was added to the chain) is used.
+
+        """
+        # todo: implement output_signal_unit features as described in docstring
+        input_signal_source = self._check_and_get_node_name(input_signal_source)
+        input_signal = self._graph.nodes[input_signal_source]["signal"]
+
+        type_tf = f"{input_signal.signal_type[0]}{output_signal_type[0]}".upper()
+
+        transformation = SignalTransformation(name, error, func, io_types=type_tf)
+
+        self.add_transformation(transformation, data, input_signal_source)
+
+    def add_transformation_from_equipment(
+        self,
+        equipment: GenericEquipment,
+        data=None,
+        input_signal_source: str = None,
+        transformation_name=None,
+    ):
+        """Add a transformation from a piece of equipment.
+
+        Parameters
+        ----------
+        equipment :
+            The equipment that contains the transformation
+        data :
+            A set of measurement data that is associated with the output signal of the
+            transformation
+        input_signal_source :
+            The source of the signal that should be used as input of the transformation.
+            If `None` is provided, the name of the last added transformation (or the
+            source, if no transformation was added to the chain) is used.
+        transformation_name :
+            In case the provided piece of equipment contains multiple transformations,
+            this parameter can be used to select one by name. If it only contains a
+            single transformation, this parameter can be set to ´None´ (default)
+
+        """
+        if len(equipment.data_transformations) > 1:
+            if transformation_name is None:
+                raise ValueError(
+                    "The equipment has multiple transformations. Specify the "
+                    "desired one by providing a 'transformation_name'."
+                )
+            transformation = equipment.get_transformation(transformation_name)
+        elif len(equipment.data_transformations) == 1:
+            transformation = equipment.data_transformations[0]
+        else:
+            raise ValueError("The equipment does not have a transformation.")
+
+        input_signal_source = self._check_and_get_node_name(input_signal_source)
+        self.add_transformation(transformation, data, input_signal_source)
+        self._graph.edges[(input_signal_source, self._prev_added_signal)][
+            "equipment"
+        ] = equipment
+
+    def add_signal_data(self, data: "TimeSeries", signal_source: str = None):
```

Comment:
From going through the code I am not sure if `Signal.data` is supposed to be a `TimeSeries` or `xr.DataArray` (or either?)